src/prefect/client/orchestration.py

```diff
@@ -3143,16 +3154,24 @@ async def read_automations(self) -> List[Automation]:
         response.raise_for_status()
         return pydantic.parse_obj_as(List[Automation], response.json())
 
-    async def find_automation(self, id_or_name: str) -> Optional[Automation]:
-        try:
-            id = UUID(id_or_name)
-        except ValueError:
-            id = None
+    async def find_automation(
```

Comment:
`get` is a more common verb for this, `find` is nonstandard

Reply:
PR up here: https://github.com/PrefectHQ/prefect/pull/12850

Reply:
Merged, so we could feel free to use those methods for reading (`read_automation_by_name`, `read_automation`)

Reply:
Ah I see; nah we can keep `find`, but I do think separating out the kwargs is more sensible so that it's `find(id: UUID = None, name: str = None)` with the appropriate kwarg handling.  I'd rather do that than have more methods with long names to keep the surface area tight.

Reply:
I agree with separating it out, but also that the rest of our methods in the client use the above convention of `read_x_by_name`, so it would probably be less cognitive overhead for the user to remember.

Reply:
That's a fair point -- I think having two named methods only makes sense if we expect `name` to be non-unique (so the return value is of a different type, list vs. item) in which case it'd be `read_automations_by_name` vs. `read_automation`

Reply:
We do expect that `name` could be non-unique, or rather we have constraints on it. I've opened a new PR here that I think makes the changes you mentioned: https://github.com/PrefectHQ/prefect/pull/12884

Reply:
Update: Merged! (thanks to Guidry for the quick review)